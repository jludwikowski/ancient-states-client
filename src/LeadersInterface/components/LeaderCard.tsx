import React from "react";
import Building from "../../Models/Building";
import Leade from "../../Models/Leader";

type Props = {
    leader: Leade,
    onDismiss(id: number): void;
} & React.HTMLAttributes<HTMLDivElement>;

export const LeaderCard: React.FunctionComponent<Props> = props => (
    <div {...props}>
        <img src={props.leader.imageUrl} />
        <div>{props.leader.name}</div>
        <div>{props.leader.cunning}</div>
        <div>{props.leader.leadership}</div>
        <div>{props.leader.wisdom}</div>
        <div>{props.leader.willpower}</div>
        <button className="btn btn-outline-secondary" onClick={() => props.onDismiss(props.leader.id)}>
            Dismiss This Leader
        </button>
    </div>
)
