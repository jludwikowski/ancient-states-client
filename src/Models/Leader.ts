import {Entity} from "./Entity";

export default interface Leader extends Entity {
    leadership: number,
    wisdom: number,
    cunning: number,
    willpower: number,
    imageUrl: ""
}
