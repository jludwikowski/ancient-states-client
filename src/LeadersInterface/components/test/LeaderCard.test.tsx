import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import Leader from "../../../Models/Leader";
import { LeaderCard } from '../LeaderCard';

describe('LeaderCard', () => {

    const leader:Leader = {
        name: 'Alcibiades',
        description: 'All fall for Alcibiades',
        leadership: 100,
        wisdom: 0,
        cunning: 100,
        willpower: 0,
        imageUrl: '',
    };

    const onDismiss = (id:number) => {
        console.log('On onDismiss');
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<LeaderCard leader={leader} onDismiss={onDismiss} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<LeaderCard leader={leader} onDismiss={onDismiss} />);
        expect(component).toMatchSnapshot();
    });
});
