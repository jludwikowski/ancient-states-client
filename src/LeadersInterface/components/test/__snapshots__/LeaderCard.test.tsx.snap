// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderCard Should match snapshot 1`] = `
<div
  leader={
    Object {
      "cunning": 100,
      "description": "All fall for Alcibiades",
      "imageUrl": "",
      "leadership": 100,
      "name": "Alcibiades",
      "willpower": 0,
      "wisdom": 0,
    }
  }
  onDismiss={[Function]}
>
  <img
    src=""
  />
  <div>
    Alcibiades
  </div>
  <div>
    100
  </div>
  <div>
    100
  </div>
  <div>
    0
  </div>
  <div>
    0
  </div>
  <button
    className="btn btn-outline-secondary"
    onClick={[Function]}
  >
    Dismiss This Leader
  </button>
</div>
`;
